Update Eventing documentation (#12765)
kyma-project/kyma: docs/05-technical-reference/evnt-01-event-types.md

---
title: Event types
---

Eventing supports the following event types:
- [Cloud Events](https://cloudevents.io/) - they use a specification for describing event data in a common way.
- legacy events - they are converted to Cloud Events by [Event Publisher Proxy](./00-architecture/evnt-01-architecture.md#event-publisher-proxy), which also adds a `sap.kyma.custom` prefix.

> ## Event form

Reviewer: (nit) In code, its referred to as `Event Type Format`, or maybe we can refer it as `Event Name Format`?

> For a Subscription Custom Resource, the fully qualified event type takes the sample form of `sap.kyma.custom.commerce.order.created.v1` or `sap.kyma.custom.commerce.Account.Root.Created.v1`.

Reviewer: (nit) maybe use "fully qualified event `name`"

The event type is composed of the following components:
- Prefix: `sap.kyma.custom`
- Application: `commerce`
- Event: can have two or more segments separated by `.`; for example, `order.created` or `Account.Root.Created`
- Version: `v1`

For publishers, the event type takes this sample form:
- `order.created` or `Account.Root.Created` for legacy events coming from the `commerce` application
- `sap.kyma.custom.commerce.order.created.v1` or `sap.kyma.custom.commerce.AccountRoot.Created.v1` for Cloud Events

> ## Event name cleanup

Author: Maybe a better name for the whole document is "Event names"? The types are linked with the names because there is a different name for different type. It's not exactly troubleshooting, just providing information about an automatic process in Eventing (making sure there are no hyphens in event names). I can also split the info into two documents? Thank you for the comment 👍

> ## Event name cleanup

Author: I renamed the document to "Event names", I hope that is okay.

In some cases, Eventing needs to modify the event name before dispatching an event. This is done in order to conform to Cloud Event specifications.

> If the event contains more than two segments, Eventing combines them into two segments when creating the underlying Eventing infrastructure. For example, `Account.Root.Created` becomes `AccountRoot.Created`.

Reviewer: (nit) maybe use "If the event `name` contains".

> In case the Application name contains `-` or `.`, the underlying Eventing services uses a clean name with alphanumeric characters only. (For example, `system-prod` becomes `systemprod`).

Reviewer 2: ```suggestion
If the Application name contains `-` or `.`, the underlying Eventing services uses a clean name with alphanumeric characters only; for example, `system-prod` becomes `systemprod`.
```

> This could lead to a naming collision. For example, both `system-prod` and `systemprod` become `systemprod`.

Author: There is no specific error message, but Kyma may not work as expected.
